CMakeLists.txt

```diff
@@ -68,6 +68,7 @@ SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/find-external/CppAD/" ${CMAKE_M
 ADD_PROJECT_DEPENDENCY(pinocchio 2.6.3 REQUIRED PKG_CONFIG_REQUIRES "pinocchio >= 2.6.3")
 ADD_PROJECT_DEPENDENCY(example-robot-data 3.7.0 REQUIRED PKG_CONFIG_REQUIRES "example-robot-data >= 3.7.0")
 ADD_OPTIONAL_DEPENDENCY("scipy")
+ADD_OPTIONAL_DEPENDENCY(ipopt)
 
```

Comment:
To enable Ipopt as an optional library, we need to add a definition that we'll pass to the library.
The code would be something like this

```
IF(IPOPT_FOUND)
  ADD_DEFINITIONS(-DCROCODDYL_WITH_IPOPT)
ENDIF()
```

Then, we'll use this macro (in the c++ files) to include or not the Ipopt files. 
See this example here: https://github.com/loco-3d/crocoddyl/pull/1020/files#r777798630

I would try to mark all the places where we need to use this macro.